pipcompilemulti/features/upgrade.py

```diff
@@ -78,10 +80,19 @@ def reset(self):
         """Clear cached packages."""
         self._env_packages_cache = {}
 
+    @property
+    def package_specs(self):
+        """List of package specs to upgrade."""
+        return self.value or []
+
     @property
     def package_names(self):
         """List of package names to upgrade."""
-        return self.value or []
+        def name_from_spec(name):
+            match = re.match(r'(?P<package>[a-z0-9-_.]+)', name)
```

Comment:
Let's move regex compilation to a class attribute, like done here: https://github.com/peterdemin/pip-compile-multi/blob/master/pipcompilemulti/dependency.py#L50

Reply:
3b471d8bd0013db7b845f71dfe0de6d8fef2a1b5